tests/aircall/test_async_funcs.py

```diff
@@ -0,0 +1,212 @@
+"""Module containing tests with fake server"""
+# from aiohttp import web
+import pytest
+
+import tests.general_helpers as helpers
+from toucan_connectors.aircall.aircall_connector import fetch_page
+
+fetch_fn_name = 'toucan_connectors.aircall.aircall_connector.fetch'
+
+# we want this module's code to be checked against Python3.8
+# versions prior to Python3.8 don't handle mocks same way for async functions
+PY_VERSION_TO_CHECK = (3, 8)
+
+is_py_version_older = helpers.check_py_version(PY_VERSION_TO_CHECK)
+
+
+async def test_fetch_page_with_no_next(mocker):
+    """Bearer version: tests that no next page returns an array of one response"""
+    dataset = 'tags'
+    fake_data = {'data': {'stuff': 'stuff'}, 'meta': {'next_page_link': None, 'current_page': 1}}
+    if is_py_version_older:
+        fake_data = helpers.build_future(fake_data)
+    fake_fetch = mocker.patch(fetch_fn_name, return_value=fake_data)
+    result = await fetch_page(dataset, [], {}, 10, 0)
+    assert len(result) == 1
+    first_dict = result[0]
+    assert first_dict.get('data') is not None
+    assert first_dict.get('meta') is not None
+
+    if is_py_version_older:
+        fake_fetch.assert_called_once_with(
+            'https://proxy.bearer.sh/aircall_oauth/tags?per_page=50&page=1', {}
+        )
+    else:
+        fake_fetch.assert_awaited_once_with(
+            'https://proxy.bearer.sh/aircall_oauth/tags?per_page=50&page=1', {}
+        )
+
+
+async def test_fetch_page_with_next_page(mocker):
+    """Test fetch_page to see multiple pages"""
+    dataset = 'calls'
+    data_list = [
+        {
+            'data': {'stuff': 'stuff'},
+            'meta': {'next_page_link': '/calls?page=2', 'current_page': 1},
+        },
+        {'data': {'stuff': 'stuff'}, 'meta': {'next_page_link': None, 'current_page': 2}},
+    ]
+    if is_py_version_older:
+        data_list = [helpers.build_future(item) for item in data_list]
+    fake_fetch = mocker.patch(fetch_fn_name, side_effect=data_list)
+
+    # limit is 10 and run is 0 i.e. this is the first run
+    # want to see that, despite the limit being set to 10,
+    # only two objects are sent in the array because no next_page_link
+    # on second object means that there is no more to be fetched
+    fake_res = await fetch_page(dataset, [], {}, 10, 0)
+    assert len(fake_res) == 2
+    if is_py_version_older:
```

Comment:
Since you use this pattern several times, you could wrap it in a single helper function and simplify code:

```python
def assert_awaited_with(mock, expected_params, expected_count=None):
    if is_py_version_older:
        if expected_count is not None:
            assert mock.call_count == expected_count
        mock.assert_called_with(*expected_params)
   # ...
```

and then

```suggestion
    helpers.assert_awaited_with(fake_fetch, [...], expected_count=2)
```